cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -759,6 +763,20 @@ impl ClientTrait for CatalogClient {
         });
         Ok(all_narinfo_available)
     }
+
+    #[instrument(skip_all)]
+    async fn get_base_catalog_info(&self) -> Result<BaseCatalogInfo, CatalogClientError> {
+        self.client
+            .get_base_catalog_api_v1_catalog_info_base_catalog_get()
+            .await
+            .map_err(|e| match e {
+                APIError::ErrorResponse(err) => {
+                    CatalogClientError::APIError(APIError::ErrorResponse(err))
+                },
+                _ => CatalogClientError::APIError(e),
+            })
```

Comment:
198b401641eef2bb3ce5f437de288f0c8f46a9c2 added an extension trait to handle these:
```suggestion
            .map_api_error()
            .await
```